File: LinkedList/Singly/append.js
class Node {
  constructor(data) {
    this.data = data;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
    this.tail = null;
    this.length = 0;
  }

  prepend(data) {
    if (data === undefined || data === null) {
      throw new Error("Valid Data Required");
    }

    const newNode = new Node(data);

    newNode.next = this.head;
    this.head = newNode;
    this.length++;

    if (this.length === 1) {
      this.tail = newNode;
    }

    return true;
  }

  append(data) {
    if (data === undefined || data === null) {
      throw new Error("Valid Data Required");
    }

    if (this.length === 0) {
      this.prepend(data);
      return true;
    }

    const newNode = new Node(data);

    this.tail.next = newNode;
    this.tail = newNode;
    this.length++;

    return true;
  }
}

File: LinkedList/Singly/Python/Insertion/append.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None
        self.length = 0

    def prepend(self,data):

        node = Node(data)
        node.next = self.head
        self.head = node
        self.length += 1

        if self.tail is None:
            self.tail = node
        return True
    
    def append(self, data):

        if self.head is None:
            self.prepend(data)
            return True
        
        node = Node(data)

        self.tail.next = node
        self.tail = node
        self.length += 1

        return True
    
    def print(self):

        current = self.head
        result = ""

        while current:
            result += f"{current.data}" + ("->" if current.next else "")
            current = current.next

        print(result)


if __name__ == "__main__":

    linked_list = LinkedList()

    linked_list.append(10)
    linked_list.append(20)
    linked_list.append(30)
    linked_list.append(40)
    linked_list.append(50)

    linked_list.print()